src/sinks/new_relic/model.rs

```diff
@@ -254,31 +254,47 @@ impl TryFrom<Vec<Event>> for LogsApiModel {
 
     fn try_from(buf_events: Vec<Event>) -> Result<Self, Self::Error> {
         let mut num_non_log_events = 0;
+        let mut num_non_object_events = 0;
 
-        let logs_array: Vec<HashMap<KeyString, Value>> = buf_events
+        let logs_array: Vec<ObjectMap> = buf_events
             .into_iter()
             .filter_map(|event| {
                 let Some(log) = event.try_into_log() else {
                     num_non_log_events += 1;
                     return None;
                 };
 
-                let mut log_model = KeyValData::new();
-                for (k, v) in log.convert_to_fields() {
-                    log_model.insert(k, v.clone());
-                }
-                if log.get(event_path!("message")).is_none() {
-                    log_model.insert("message".into(), Value::from("log from vector".to_owned()));
+                // We convert the log event into a logs API model simply by transmuting the type
+                // wrapper and dropping all arrays, which are not supported by the API. We could
+                // flatten out the keys, as this is what New Relic does internally, and we used to
+                // do that, but the flattening iterator accessed through
+                // `LogEvent::convert_to_fields` adds quotes to dotted fields names, which produces
+                // broken attributes in New Relic, and nesting objects is actually a (slightly) more
+                // efficient representation of the key names.
+                let (value, _metadata) = log.into_parts();
+                let Some(mut obj) = value.into_object() else {
```

Comment:
Rustlang features too cool for GH syntax highlighting. GH can't handle the coolness